### PR TITLE
[MRG] ENH: add verbose param to to_data_frame methods

### PR DESCRIPTION
**Pull request by sappelhoff**
When calling `.to_data_frame` on a number of objects in a loop, the output can get quite verbose. Here I add the `verbose` decorator to the function calls where appropriate (I hope).

The logging that gets too spammy happens in `_set_pandas_dtype`.

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -2073,10 +2073,11 @@ def equalize_event_counts(self, event_ids=None, method='mintime'):
         # actually remove the indices
         return self, indices
 
+    @verbose
     @fill_doc
```
Comment (larsoner) [inline mne/epochs.py:2077]: `@verbose` does a `@fill_doc` under the hood so you can remove this
```suggestion
    @verbose
```

File: mne/epochs.py [hunk 2]
```diff
@@ -2095,6 +2096,7 @@ def to_data_frame(self, picks=None, index=None,
         %(df_copy)s
         %(df_longform_epo)s
         %(df_time_format)s
+        %(verbose)s
 
             .. versionadded:: 0.20
```
Comment (larsoner) [inline mne/epochs.py:2101]: Keep the versionadded with the `df_time_format` param
```suggestion

            .. versionadded:: 0.20
        %(verbose)s

            .. versionadded:: 1.0
```
You could add one for `verbose`, too, but we sometimes don't bother for that `kwarg`

File: mne/epochs.py [hunk 3]
```diff
@@ -2130,7 +2132,8 @@ def to_data_frame(self, picks=None, index=None,
         assert all(len(mdx) == len(mindex[0]) for mdx in mindex)
         # build DataFrame
         df = _build_data_frame(self, data, picks, long_format, mindex, index,
-                               default_index=['condition', 'epoch', 'time'])
+                               default_index=['condition', 'epoch', 'time'],
+                               verbose=verbose)
```
Comment (larsoner) [inline mne/epochs.py:2136]: There should be no need to pass this param to `_build_data_frame` based on how the magic of `verbose` works
```suggestion
                               default_index=['condition', 'epoch', 'time'])
```

File: mne/evoked.py [hunk 4]
```diff
@@ -778,10 +778,11 @@ def get_peak(self, ch_type=None, tmin=None, tmax=None,
 
         return out
 
+    @verbose
```
Comment (larsoner) [inline mne/evoked.py:781]: same stuff in this file and the others

**Review by larsoner — CHANGES_REQUESTED** [review]

**Review by sappelhoff — COMMENTED** [review]
Thanks @larsoner I think I addressed all your points. I added one thing: I thought it was a good idea to make the parameters before "verbose" keyword only, so that we can more easily add additional kwargs in the future and keep verbose as the last kwarg without breaking backward compatibility. WDYT about that?

**Comment by larsoner** [comment]
Thanks @sappelhoff !

**Pull request merged**